java/statestore/src/test/java/sleeper/statestore/s3/S3StateStoreIT.java

```diff
@@ -1181,8 +1189,9 @@ public void shouldNotUpdatePartitionsIfLeafStatusChanges() throws IOException, S
 
         // When / Then
         //  - Attempting to split something that has already been split should fail
+        Partition finalParentPartition = parentPartition;
```

Comment:
This statement and the variable name look a bit strange, instead of needing to do this, you could assign the result of `parentPartition.toBuilder().build()` above to a variable named something like `parentAfterSplit` like this:

```java
Partition parentAfterSplit = parentPartition.toBuilder()
    .leafPartition(false)
    .childPartitionIds(Arrays.asList("child1", "child2"))
    .build();

assertThatThrownBy(() -> dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(parentAfterSplit, childPartition1, childPartition2))
```

Same applies to the tests after this one